src/include/migraphx/op/allocate.hpp

```diff
@@ -62,26 +62,38 @@ struct allocate
 
     shape compute_shape(const std::vector<shape>& inputs) const
     {
-        if(s != shape())
+        if(s)
```

Comment:
```suggestion
        if(s.has_value())
```
Minor Suggestion : For better readability